Add carousel pattern

mcking65 opened the pull request:
Adds a design pattern section for carousels to resolve issue #43.

#### Preview Pattern in Feature Branch

[Carousel pattern in issue43-add-carousel-pattern branch](https://raw.githack.com/w3c/aria-practices/issue43-add-carousel-pattern/aria-practices.html#carousel)


<!--
    This comment and the below content is programatically generated.
    You may add a comma-separated list of anchors you'd like a
    direct link to below (e.g. #idl-serializers, #idl-sequence):

    Don't remove this comment or modify anything below this line.
    If you don't want a preview generated for this pull request,
    just replace the whole of this comment's content by "no preview"
    and remove what's below.
-->
***
<a href="https://pr-preview.s3.amazonaws.com/w3c/aria-practices/pull/957.html" title="Last updated on Jan 15, 2019, 12:54 AM UTC (6f2f95e)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/aria-practices/957/08e5fe0...6f2f95e.html" title="Last updated on Jan 15, 2019, 12:54 AM UTC (6f2f95e)">Diff</a>

shirsha commented:
I like the updated definition but  the word "horizontal scrolling" is misleading as carousels do not use a scroll bar to navigate through the slides.

mcking65 commented:
@charmarkk, many, many thanks for the review!!!

>"...image carousels where each slide contains nothing more than a single image are common." 
>-Add "**most** common" instead of "common."

Do we know this for sure? Does adding "most" enhance the meaning? If they are most common, extremely common, or just very common, they are nonetheless common. I'm happy to add "most" if we really know it to be the case, and it meaningfully strengthens the description.

>"Basic: Has rotation , previous slide,..."
>-Remove space after rotation, "Has rotation, ..."

Fixed in commit 8856024.

>(Last bullet under Basic Carousel Elements) "... does not contain the word "slide".
>-End punctuation goes inside of quotation marks.

Fixed in commit 8856024.

>(Under Grouped Carousel Elements) "Choose slide to display".
>-End punctuation goes inside of quotation marks.

Fixed in commit 8856024.

BTW, when doing a review, you can put your comments right on the line of the file that needs to be changed. Go to the files tab and click the add comment button by the line the needs changing. That is one way to start a review. You can glob a bunch of comments like that together into a review.

mcking65 commented:
shirsha commented:

>I like the updated definition but the word "horizontal scrolling" is misleading as carousels do not use a scroll bar to navigate through the slides.

Thank you Siri. Removed term horizontal scrolling. Please have a look at the latest.

charmarkk commented:
@mcking65 TIL, my thanks! 

As for the question of adding “most” or not, I mostly mentioned that because it read better. If it’s a question of veracity, however, I’m happy leaving it as is.

jongund commented:
@mcking65 The carousel design pattern looks good to me.  I have updated the carousel example to include aria-roledescription on the container section element.

annabbott commented on aria-practices.html:
```diff
@@ -477,7 +477,171 @@ <h4>WAI-ARIA Roles, States, and Properties</h4>
+            Optionally, an element wrapping the set of slide elements has <a href="#aria-atomic" class="property-reference">aria-atomic</a> set to <code>false</code>
+            and <a href="#aria-live" class="property-reference">aria-live</a> set to:
+            <ul>
+              <li><code>off</code>: ifthe the carousel is automatically rotating.
```
Add space between "if" and "the"
